origin_ca.go

```diff
@@ -221,6 +221,11 @@ func OriginCARootCertificate(algorithm string) ([]byte, error) {
 		return nil, errors.Wrap(err, "HTTP request failed")
 	}
 	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
```

Comment:
part of the issue here is that a few lines above, we're using `http.Get`, not our client which automatically handles this.